### PR TITLE
[AIRFLOW-4724] Make params dict to be templated for operators

### PR DESCRIPTION
**Pull request by srivignessh**
Make sure you have checked _all_ steps below.

### Description

I am proposing this change to fix the following bug
```
EXEC_TIMESEPOCH =  "{{ execution_date.strftime('%s') }}"
gcs_export_uri_template_filename = 'product_dwh-' + EXEC_TIMESEPOCH + '.csv'
upload_file_ftp_op = BashOperator(
   task_id='upload_file_ftp_task',
   params={'filename':gcs_export_uri_template_filename},
   bash_command="python3.6 /home/ubuntu/airflow/scripts/ranker.py  '{{ params.filename }}'  " ,
  dag=dag)
```

Gives:
```
python3.6 /home/ubuntu/airflow/scripts/ranker.py  'product_dwh-{{ execution_date.strftime('%s') }}.csv'
```

Expected:
```
python3.6 /home/ubuntu/airflow/scripts/ranker.py  product_dwh-1433142000.csv
```

I have noticed that params was available in jinja context but it wasn't templated as it was nested. 
Hence, running the result once again in the same context obtained the required result.

I request the reviewers for further comments.

If this approach is good to go. I will append the commit with test cases.

Thanks,
Srivignessh

### CODE REVIEW

File: airflow/models/baseoperator.py [hunk 1]
```diff
@@ -645,7 +663,9 @@ def render_template_from_field(self, attr, content, context, jinja_env):
         """
         rt = self.render_template
         if isinstance(content, str):
-            result = jinja_env.from_string(content).render(**context)
+            result = content
+            while(self.is_template(result, jinja_env)):
```
Comment (BasPH) [inline airflow/models/baseoperator.py:667]: Why `while`?

**Review by BasPH — CHANGES_REQUESTED** [review]
I'd be okay with making `params` templatable by default, but I'm -1 on the `is_template` check. It appears overly complex and unnecessary to me.

**Comment by BasPH** [comment]
Answering your question: `params` is not templated by default, nor by the BashOperator. You can check in any operator which attributes are templated by looking for the `template_fields` attribute. In the BashOperator, you'll see the `bash_command` in there and thus you could run your operator as such:

```python
upload_file_ftp_op = BashOperator(
   task_id='upload_file_ftp_task',
   bash_command="python3.6 /home/ubuntu/airflow/scripts/ranker.py product_dwh-{{ execution_date.strftime('%s') }}.csv" ,
   dag=dag,
)
```

**Comment by BasPH** [comment]
Closing for now, feel free to ask anything on the Airflow Slack: https://apache-airflow-slack.herokuapp.com.

**Comment by OmerJog** [comment]
@basPH please refer to the Jira https://issues.apache.org/jira/browse/AIRFLOW-4724 this has been discussed at length in https://apache-airflow.slack.com/archives/CCQ7EGB1P/p1559134166151100
it require a fix. What you suggested is already given in the ticket. It's not a solution it's a work around.

BaseOperator specifically says:
https://github.com/apache/airflow/blob/master/airflow/models/baseoperator.py#L367 
`        self.params = params or {}  # Available in templates!
`

So it needs to be supported as is.

**Comment by ashb** [comment]
`# Available in templates` means you can do `{{ params.foo }}`, not that the foo key you put in the params dict _is_ templated.

**Comment by srivignessh** [comment]
@ashb I agree with your statement. Thanks for making it clear.

**Pull request closed without merging**